svc.go

```diff
@@ -123,7 +126,18 @@ func (as *autoNATService) handleDial(p peer.ID, obsaddr ma.Multiaddr, mpi *pb.Me
 		}
 
 		if as.config.dialPolicy.skipDial(addr) {
-			continue
+			if manet.IsPrivateAddr(addr) {
+				addrcodename, addrport, err := as.extractPort(addr)
+				if err == nil && addrcodename == obscodename && addrport != obsport { //make a new address
+					obsportstr := fmt.Sprintf("/%s/%s", obscodename, obsport)
+					addrportstr := fmt.Sprintf("/%s/%s", addrcodename, addrport)
+					addr, err = ma.NewMultiaddr(strings.Replace(obsaddr.String(), obsportstr, addrportstr, -1)) //replace the private addr
+				} else {
```

Comment:
`IsPrivateAddr` will also include the `127.0.0.1` addresses, which probably aren't useful. I wonder if there's any more selective way to filter ports we care about.